storage/watchable_store.go

```diff
@@ -185,7 +186,9 @@ func (s *watchableStore) watch(key []byte, prefix bool, startRev int64, ch chan<
 
 	k := string(key)
 	if startRev == 0 {
-		s.synced[k] = append(s.synced[k], wa)
+		if err := unsafeAddWatching(&s.synced, k, wa); err != nil {
+			log.Panicf("error unsafeAddWatching (%v) for key %s", err, k)
```

Comment:
after offline discussion, we decided to still return the error for easy testing.